include/util/conditional_restrictions.hpp

```diff
@@ -0,0 +1,118 @@
+#ifndef CONDITIONAL_RESTRICTIONS_HPP
+#define CONDITIONAL_RESTRICTIONS_HPP
+
+//#define BOOST_SPIRIT_DEBUG
+
+#include <boost/fusion/include/adapt_struct.hpp>
+#include <boost/spirit/include/phoenix.hpp>
+#include <boost/spirit/include/qi.hpp>
+
+namespace osrm
+{
+namespace util
+{
+
+// Helper functions for OSM conditional restrictions
+// http://wiki.openstreetmap.org/wiki/Conditional_restrictions
+// Consitional restrictions is a vector of ConditionalRestriction
+// with a restriction value and a condition string
+struct ConditionalRestriction
+{
+    std::string value;
+    std::string condition;
+};
+
+// TODO: remove debug
+// inline std::ostream &operator<<(std::ostream &stream, const ConditionalRestriction &restriction)
+// {
+//     return stream << restriction.value << "=" << restriction.condition;
+// }
```

Comment:
I will keep it as is  in 
```
#ifndef NDEBUG
// Debug output stream operators for use with BOOST_SPIRIT_DEBUG
...
#endif
```

It is only needed for `BOOST_SPIRIT_DEBUG` that is extremely helpful for complex cases